File: py/sort/shuffle/InsideOut.py
#!/usr/bin/env python
# -*- coding:gbk -*-

import random
from typing import List

class InsideOut:
    def shuffle(self, nums: List[int]) -> List[int]:
        result = list(nums)
        for i in range(len(nums)):
            index = random.randrange(0, i + 1)
            result[i], result[index] = result[index], nums[i]
        return result

File: java/sort/shuffle/InsideOut.java
package sort.shuffle;

import java.util.Random;

public class InsideOut {
    private static final Random _random = new Random();

    public int[] shuffle(int[] nums) {
        int[] result = nums.clone();
        for (int i = 0; i < nums.length; ++i) {
            int index = _random.nextInt(i + 1);
            result[i] = result[index];
            result[index] = nums[i];
        }
        return result;
    }
}
